### PR TITLE
[FLINK-27185][connectors] Convert connector-kafka module to assertj

### PR DESCRIPTION
**Pull request by alpreu**
Cherry-picked commits from https://github.com/apache/flink/pull/19660 to get the individual changes merged faster

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaSinkBuilderTest.java [hunk 1]
```diff
@@ -73,9 +72,9 @@ public void testPropertyHandling() {
                         .setKafkaProducerConfig(testConf)
                         .setProperty("k2", "correct"),
                 p -> {
-                    Arrays.stream(DEFAULT_KEYS).forEach(k -> assertTrue(k, p.containsKey(k)));
-                    assertEquals("v1", p.get("k1"));
-                    assertEquals("correct", p.get("k2"));
+                    Arrays.stream(DEFAULT_KEYS).forEach(k -> assertThat(p).containsKey(k));
+                    assertThat(p.get("k1")).isEqualTo("v1");
+                    assertThat(p.get("k2")).isEqualTo("correct");
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaSinkBuilderTest.java:77]: Here `containsEntry` could be used + joined
```suggestion
                    assertThat(p).containsEntry("k1", "v1").containsEntry("k2", "correct");
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/initializer/OffsetsInitializerTest.java [hunk 2]
```diff
@@ -65,36 +63,38 @@ public void testEarliestOffsetsInitializer() {
...
 
     @Test
     public void testLatestOffsetsInitializer() {
         OffsetsInitializer initializer = OffsetsInitializer.latest();
         List<TopicPartition> partitions = KafkaSourceTestEnv.getPartitionsForTopic(TOPIC);
         Map<TopicPartition, Long> offsets = initializer.getPartitionOffsets(partitions, retriever);
-        assertEquals(partitions.size(), offsets.size());
-        assertTrue(offsets.keySet().containsAll(partitions));
+        assertThat(offsets).hasSize(partitions.size());
+        assertThat(offsets.keySet().containsAll(partitions)).isTrue();
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/initializer/OffsetsInitializerTest.java:81]:
```suggestion
        assertThat(offsets).hasSameSizeAs(partitions);
        assertThat(offsets.keySet()).containsAll(partitions);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/initializer/OffsetsInitializerTest.java [hunk 3]
```diff
@@ -65,36 +63,38 @@ public void testEarliestOffsetsInitializer() {
...
+        assertThat(initializer.getAutoOffsetResetStrategy()).isEqualTo(OffsetResetStrategy.LATEST);
     }
 
     @Test
     public void testCommittedGroupOffsetsInitializer() {
         OffsetsInitializer initializer = OffsetsInitializer.committedOffsets();
         List<TopicPartition> partitions = KafkaSourceTestEnv.getPartitionsForTopic(TOPIC);
         Map<TopicPartition, Long> offsets = initializer.getPartitionOffsets(partitions, retriever);
-        assertEquals(partitions.size(), offsets.size());
+        assertThat(offsets).hasSize(partitions.size());
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/initializer/OffsetsInitializerTest.java:93]:
```suggestion
        assertThat(offsets).hasSameSizeAs(partitions);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/KafkaEnumeratorTest.java [hunk 4]
```diff
@@ -304,16 +298,15 @@ public void testKafkaClientProperties() throws Exception {
...
-            assertNotNull(clientId);
-            assertTrue(clientId.startsWith(clientIdPrefix));
-            assertEquals(
-                    defaultTimeoutMs,
-                    Whitebox.getInternalState(adminClient, "defaultApiTimeoutMs"));
-
-            assertNotNull(clientId);
-            assertTrue(clientId.startsWith(clientIdPrefix));
+            assertThat(clientId).isNotNull();
+            assertThat(clientId).startsWith(clientIdPrefix);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/KafkaEnumeratorTest.java:304]:
```suggestion
            assertThat(clientId).isNotNull().startsWith(clientIdPrefix);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/KafkaEnumeratorTest.java [hunk 5]
```diff
@@ -304,16 +298,15 @@ public void testKafkaClientProperties() throws Exception {
...
+            assertThat(Whitebox.getInternalState(adminClient, "defaultApiTimeoutMs"))
+                    .isEqualTo(defaultTimeoutMs);
+
+            assertThat(clientId).isNotNull();
+            assertThat(clientId).startsWith(clientIdPrefix);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/enumerator/KafkaEnumeratorTest.java:309]:
```suggestion
            assertThat(clientId).isNotNull().startsWith(clientIdPrefix);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java [hunk 6]
```diff
@@ -220,25 +222,25 @@ public void testPartitionsFewerThanConsumersFixedPartitions() {
 
                 List<KafkaTopicPartition> initialDiscovery =
                         partitionDiscoverer.discoverPartitions();
-                assertTrue(initialDiscovery.size() <= 1);
+                assertThat(initialDiscovery.size()).isLessThanOrEqualTo(1);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java:225]:
```suggestion
                assertThat(initialDiscovery).hasSizeLessThanOrEqualTo(1);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java [hunk 7]
```diff
@@ -314,33 +316,39 @@ public void testGrowingPartitions() {
...
+            assertThat(initialDiscoverySubtask0.size())
+                    .isLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
+            assertThat(initialDiscoverySubtask1.size())
+                    .isGreaterThanOrEqualTo(minInitialPartitionsPerConsumer);
+            assertThat(initialDiscoverySubtask1.size())
+                    .isLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
+            assertThat(initialDiscoverySubtask2.size())
+                    .isGreaterThanOrEqualTo(minInitialPartitionsPerConsumer);
+            assertThat(initialDiscoverySubtask2.size())
+                    .isLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java:330]:
```suggestion
            assertThat(initialDiscoverySubtask0)
                    .hasSizeGreaterThanOrEqualTo(minInitialPartitionsPerConsumer)
                    .hasSizeLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
            assertThat(initialDiscoverySubtask1)
                    .hasSizeGreaterThanOrEqualTo(minInitialPartitionsPerConsumer)
                    .hasSizeLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
            assertThat(initialDiscoverySubtask2)
                    .hasSizeGreaterThanOrEqualTo(minInitialPartitionsPerConsumer)
                    .hasSizeLessThanOrEqualTo(maxInitialPartitionsPerConsumer);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java [hunk 8]
```diff
@@ -507,15 +512,15 @@ public void testNonContiguousPartitionIdDiscovery() throws Exception {
         partitionDiscoverer.open();
 
         List<KafkaTopicPartition> discoveredPartitions1 = partitionDiscoverer.discoverPartitions();
-        assertEquals(2, discoveredPartitions1.size());
-        assertTrue(discoveredPartitions1.contains(new KafkaTopicPartition("test-topic", 1)));
-        assertTrue(discoveredPartitions1.contains(new KafkaTopicPartition("test-topic", 4)));
+        assertThat(discoveredPartitions1).hasSize(2);
+        assertThat(discoveredPartitions1).contains(new KafkaTopicPartition("test-topic", 1));
+        assertThat(discoveredPartitions1).contains(new KafkaTopicPartition("test-topic", 4));
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java:517]:
```suggestion
        assertThat(discoveredPartitions1)
                .hasSize(2)
                .contains(
                        new KafkaTopicPartition("test-topic", 1),
                        new KafkaTopicPartition("test-topic", 4));
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java [hunk 9]
```diff
@@ -507,15 +512,15 @@ public void testNonContiguousPartitionIdDiscovery() throws Exception {
...
 
         List<KafkaTopicPartition> discoveredPartitions2 = partitionDiscoverer.discoverPartitions();
-        assertEquals(3, discoveredPartitions2.size());
-        assertTrue(discoveredPartitions2.contains(new KafkaTopicPartition("test-topic", 0)));
-        assertTrue(discoveredPartitions2.contains(new KafkaTopicPartition("test-topic", 2)));
-        assertTrue(discoveredPartitions2.contains(new KafkaTopicPartition("test-topic", 3)));
+        assertThat(discoveredPartitions2).hasSize(3);
+        assertThat(discoveredPartitions2).contains(new KafkaTopicPartition("test-topic", 0));
+        assertThat(discoveredPartitions2).contains(new KafkaTopicPartition("test-topic", 2));
+        assertThat(discoveredPartitions2).contains(new KafkaTopicPartition("test-topic", 3));
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractPartitionDiscovererTest.java:523]:
```suggestion
        assertThat(discoveredPartitions2)
                .hasSize(3)
                .contains(
                        new KafkaTopicPartition("test-topic", 0),
                        new KafkaTopicPartition("test-topic", 2),
                        new KafkaTopicPartition("test-topic", 3));
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestUtils.java [hunk 10]
```diff
@@ -99,12 +99,12 @@ public static void comparedWithKeyAndOrder(
             actualData.computeIfAbsent(key, k -> new LinkedList<>()).add(row);
         }
         // compare key first
-        assertEquals("Actual result: " + actual, expectedData.size(), actualData.size());
+        assertThat(actualData).as("Actual result: " + actual).hasSize(expectedData.size());
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestUtils.java:102]:
```suggestion
        assertThat(actualData).as("Actual result: " + actual).hasSameSizeAs(expectedData);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java [hunk 11]
```diff
@@ -275,20 +274,18 @@ public void testRestoreFromEmptyStateWithPartitions() throws Exception {
...
         // assert that there are partitions and is identical to expected list
-        assertTrue(consumerFunction.getSubscribedPartitionsToStartOffsets() != null);
-        assertTrue(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty());
-        assertEquals(
-                expectedSubscribedPartitionsWithStartOffsets,
-                consumerFunction.getSubscribedPartitionsToStartOffsets());
+        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets()).isNotNull();
+        assertThat(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty()).isTrue();
+        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets())
+                .isEqualTo(expectedSubscribedPartitionsWithStartOffsets);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java:280]: Under the hood `isNotEmpty()`, `isEmpty()` check for is not null, also could be joined
```suggestion
        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets())
                .isNotEmpty()
                .isEqualTo(expectedSubscribedPartitionsWithStartOffsets);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java [hunk 12]
```diff
@@ -275,20 +274,18 @@ public void testRestoreFromEmptyStateWithPartitions() throws Exception {
...
 
         // the new partitions should have been considered as restored state
-        assertTrue(consumerFunction.getRestoredState() != null);
-        assertTrue(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty());
+        assertThat(consumerFunction.getRestoredState()).isNotNull();
+        assertThat(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty()).isTrue();
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java:284]:
```suggestion
        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets()).isNotEmpty();
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java [hunk 13]
```diff
@@ -275,20 +274,18 @@ public void testRestoreFromEmptyStateWithPartitions() throws Exception {
...
         for (Map.Entry<KafkaTopicPartition, Long> expectedEntry :
                 expectedSubscribedPartitionsWithStartOffsets.entrySet()) {
-            assertEquals(
-                    expectedEntry.getValue(),
-                    consumerFunction.getRestoredState().get(expectedEntry.getKey()));
+            assertThat(consumerFunction.getRestoredState().get(expectedEntry.getKey()))
+                    .isEqualTo(expectedEntry.getValue());
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java:288]:
```suggestion
            assertThat(consumerFunction.getRestoredState()).containsEntry(
                    expectedEntry.getKey(),
                    expectedEntry.getValue());
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java [hunk 14]
```diff
@@ -325,15 +322,16 @@ public void testRestore() throws Exception {
...
         // assert that there are partitions and is identical to expected list
-        assertTrue(consumerFunction.getSubscribedPartitionsToStartOffsets() != null);
-        assertTrue(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty());
+        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets()).isNotNull();
+        assertThat(!consumerFunction.getSubscribedPartitionsToStartOffsets().isEmpty()).isTrue();
 
         // on restore, subscribedPartitionsToStartOffsets should be identical to the restored state
-        assertEquals(PARTITION_STATE, consumerFunction.getSubscribedPartitionsToStartOffsets());
+        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets())
+                .isEqualTo(PARTITION_STATE);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseMigrationTest.java:330]: All these lines could be joined in something like these
```java
        assertThat(consumerFunction.getSubscribedPartitionsToStartOffsets())
                .isNotEmpty()
                // on restore, subscribedPartitionsToStartOffsets should be identical to the restored state
                .isEqualTo(PARTITION_STATE);
```

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java [hunk 15]
```diff
@@ -82,14 +82,16 @@ public void testKeyValueDeserializersSetIfMissing() throws Exception {
...
+        assertThat(props).containsKey(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
+        assertThat(props).containsKey(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG);
+        assertThat(
+                        props.getProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG)
+                                .equals(ByteArraySerializer.class.getName()))
+                .isTrue();
+        assertThat(
+                        props.getProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG)
+                                .equals(ByteArraySerializer.class.getName()))
+                .isTrue();
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java:94]: Is it expected to have these lines twice?

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java [hunk 16]
```diff
@@ -82,14 +82,16 @@ public void testKeyValueDeserializersSetIfMissing() throws Exception {
...
-        assertTrue(props.containsKey(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG));
-        assertTrue(props.containsKey(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG));
-        assertTrue(
-                props.getProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG)
-                        .equals(ByteArraySerializer.class.getName()));
-        assertTrue(
-                props.getProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG)
-                        .equals(ByteArraySerializer.class.getName()));
+        assertThat(props).containsKey(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
+        assertThat(props).containsKey(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG);
```
Comment (snuyanzin) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java:86]:
```suggestion
        assertThat(props)
                .containsKey(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG)
                .containsKey(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG);
```

**Review by snuyanzin — COMMENTED** [review]
In general lgtm, I left some comments

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java (hunk 15 above)
Comment (alpreu) [reply flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java:94]: Good catch,  looks like this was a broken test

**Comment by alpreu** [comment]
@flinkbot run azure

**Comment by afedulov** [comment]
LGTM

File: flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java [hunk 17]
```diff
@@ -165,13 +164,13 @@ public void testAsyncErrorRethrownOnInvoke() throws Throwable {
...
             // the next invoke should rethrow the async exception
-            Assert.assertTrue(e.getCause().getMessage().contains("artificial async exception"));
+            assertThat(e.getCause().getMessage()).contains("artificial async exception");
 
             // test succeeded
             return;
         }
 
-        Assert.fail();
+        fail("unknown failure");
```
Comment (XComp) [inline flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java:173]: nit: Those are odd ways of doing it. I guess, using the assertThrows here would be more appropriate.

**Review by XComp — APPROVED** [review]
There are a few places where we could have leveraged more of the assertj API but for now, that's good enough, I guess. 👍

**Pull request merged**